openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -182,6 +196,12 @@ def update_account_settings(requesting_user, update, username=None):
         for serializer in user_serializer, legacy_profile_serializer:
             serializer.save()
 
+        if "account_privacy" in update:
+            if account_privacy is not None:
```

Comment:
This depends on a variable set conditionally at L164.  It works, because the guard condition (`if "account_privacy" in update`) is the same, but it's fragile, because editing the code up above would break this.  Probably better to write `if update["account_privacy"] is not None` in spite of the extra dict lookup.